lib/chef/knife/google_server_create.rb

```diff
@@ -192,9 +192,11 @@ def validate_params!
           "It is required when resetting passwords on Windows hosts."
       end
 
+      raise "Please provide connection port via --connection-port." unless locate_config_value(:connection_port)
+      raise "Please provide image os type via --image-os-type." unless locate_config_value(:image_os_type)
       ui.warn("Auto-migrate disabled for preemptible instance") if preemptible? && locate_config_value(:auto_migrate)
       ui.warn("Auto-restart disabled for preemptible instance") if preemptible? && locate_config_value(:auto_restart)
-      ui.warn("[DEPRECATED] --bootstrap-protocol option is deprecated. Use --connection-protocol option instead.")  if locate_config_value(:bootstrap_protocol)
+      ui.warn("[DEPRECATED] --bootstrap-protocol option is deprecated. Use --connection-protocol option instead.") if locate_config_value(:bootstrap_protocol)
```

Comment:
Now that we require Chef 15 we may be able to use the new deprecation functionality in mixlib-cli to handle the deprecation right in the option definition. See how we did it directly in knife here: https://github.com/chef/chef/pull/8637/files